torch/lib/c10d/ProcessGroupNCCL.hpp

```diff
@@ -259,6 +249,15 @@ class ProcessGroupNCCL : public ProcessGroup {
           "FutureNCCL's value was internally set to NCCL collective's "
           "outputs or the return value of the callback.");
       value_ = std::move(value);
+
+      if (cudaEvents_ == nullptr) {
+        // Create a new cudaEvents object of size 1 that will record the current
+        // stream after callback and will be passed to the new FutureNCCL.
+        cudaEvents_ = std::make_shared<std::vector<at::cuda::CUDAEvent>>(1);
+        // In case of chained then callback calls, cudaEvents
+        // records callback's stream.
+        (*cudaEvents_)[0].record(at::cuda::getDefaultCUDAStream(deviceIndex_));
```

Comment:
ditto to the previous PR, this should be current stream instead?

Reply:
Yes, indeed I got confused here, should be fixed now.